### ISSUE TITLE
Computing AUC on epoch end [suggestions are welcomed]

### ISSUE THREAD

**Issue by francescopisu** [jrzaurin/pytorch-widedeep#69]
Hey, I'm trying to write a custom metric to allow computing AUC score after each epoch's end. I tried using torchmetric's AUC but it seems that the attribute num_classes is required and I can't set it on torchmetric's AUC.
It seems weird since for Accuracy num_classes is inferred from the shape of y_pred and I can't set num_classes on it as well.

This is where I am so far:

```
from pytorch_widedeep.metrics import Metric
from sklearn.metrics import roc_auc_score

class CustomAUC(Metric):
  def __init__(self, num_classes): # here I'm forcing the parameter on init
    if num_classes > 2:
      raise ValueError("AUC for multiclass problems is not currently supported.")
    
    self.num_classes = num_classes
    self._name = "auc"
  
  def reset(self):
    # I don't know what should be put here
    pass

  def __call__(self, y_pred, y_true):
    yp = y_pred.clone().detach().cpu().numpy()
    yt = y_true.clone().detach().cpu().numpy()
    return roc_auc_score(yt, yp)
```

Right now is working but I was wondering if there is (it surely does) a better way to write this piece of code. Suggestions are welcomed!

Thank you, awesome library

**Comment by jrzaurin** [comment]
@5uperpalo can you have a look to why the AUC in `Torchmetrics` is not working?

**Comment by 5uperpalo** [comment]
@francescopisu , just an FYI I am working on it, I will create a solution tomorrow, today I prepared a working example:
* added imbalanced multiclass dataset to datasets in library
* created [testing scenario notebook](https://github.com/jrzaurin/pytorch-widedeep/blob/pmulinka/auc/examples/15_AUC_multiclass.ipynb)

**Comment by 5uperpalo** [comment]
@francescopisu @jrzaurin please look at the proposed solution:

**Comment by francescopisu** [comment]
@5uperpalo Thank you for all this work! Yeah, I later realized that AUC class computes the area using the trapezoidal rule and that AUROC was what I was really looking for.  It is now working as expected!
